extensions/interactions/ItemSelectionInput/directives/oppia-response-item-selection-input.component.ts

```diff
@@ -40,16 +40,20 @@ export class ResponseItemSelectionInputComponent implements OnInit {
   constructor(private htmlEscaperService: HtmlEscaperService) {}
 
   ngOnInit(): void {
-    const answer = this.htmlEscaperService.escapedJsonToObj(
+    let answer = this.htmlEscaperService.escapedJsonToObj(
       this.answer
     ) as string[];
+    if (!answer) {
+      answer = [];
+    }
     const choices = this.htmlEscaperService.escapedJsonToObj(
       this.choices
     ) as { _html: string; _contentId: string }[];
 
     const choicesContentIds = choices.map(choice => choice._contentId);
-    this.responses = answer.map(
```

Comment:
These lines were replaced because when no answers are selected (when the learner is answering the questions) and the submit button is clicked, the answer array is undefined, and this causes two errors:
 - Calling `map` causes a console error since answer is not an array
 - `this.responses` is set to undefined due to the map function not working, and this also causes a bunch of console errors
 
This is fixed by separately setting `answer` to an empty array when no answer is selected (up above).

Reply:
Thanks @nithinrdy, good find!

But if you're now defining the `answer` array, why do you still need to replace these lines?

Reply:
Done, thanks for pointing this out!